src/pages/workspaces/workspace/applications/AppLauncher.js

```diff
@@ -19,16 +19,42 @@ const ApplicationLauncher = _.flow(
   Utils.forwardRefWithName('ApplicationLauncher'),
   wrapWorkspace({
     breadcrumbs: props => breadcrumbs.commonPaths.workspaceDashboard(props),
-    title: _.get('application')
+    title: _.get('application'),
+    activeTab: appLauncherTabName
   })
-)(({ namespace, name, refreshRuntimes, runtimes, persistentDisks, application, workspace }, ref) => {
+)(({ name, refreshRuntimes, runtimes, persistentDisks, application, workspace, workspace: { workspace: { googleProject, bucketName } } }, ref) => {
   const cookieReady = Utils.useStore(cookieReadyStore)
   const [showCreate, setShowCreate] = useState(false)
   const [busy, setBusy] = useState(false)
+  const [isWelderInit, setIsWelderInit] = useState(application !== 'RStudio') // We've already init Welder if app is Jupyter.
```

Comment:
Perhaps we shouldn't assume that `not 'RStudio' == 'Jupyter'`?

```suggestion
  const [isWelderInit, setIsWelderInit] = useState(application === 'Jupyter') // We've already init Welder if app is Jupyter.
```
Note: I couldn't be sure if `'Jupyter'` is the exact string for when the application is Jupyter. Please use the correct one :)

Reply:
The boolean expression here ensures we only attempt to init welder in the app launcher if the application is RStudio (isWelderInit only resolves to false `if application === RStudio`). Semantically this makes more sense to me since welder is already initialized at this point in the flow if application !== RStudio.

The way this is set up, if another application is added, there's no path via which welder will be invoked for file syncing unless explicitly added to one of the two aforementioned paths.

Reply:
I see. Then how about we rename `isWelderInit` as `shouldWelderBeSetUp` since that's really why this state is and it feels clarity is worth the verbosity in this case?

Reply:
hmm.. that sounds good to me